### PR TITLE
add topcon model connector

### PR DESCRIPTION
**Pull request by enzococca**

### CODE REVIEW

File: totalopenstation/models/topcon.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+# filename: zeiss_elta_r55.py
+# Copyright 2008 Luca Bianconi <luxetluc@yahoo.it>
+# Copyright 2008-2011 Stefano Costa <steko@iosa.it>
+
```
Comment (steko) [inline totalopenstation/models/topcon.py]: These lines should be changed to reflect the filename and the author

File: totalopenstation/models/topcon.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# <http://www.gnu.org/licenses/>.
+
+
+from . import Connector
+
+
+class ModelConnector(Connector):
+
+    def __init__(self, port):
+        Connector.__init__(self, port=port, baudrate=9600 , bytesize=8)
```
Comment (steko) [inline totalopenstation/models/topcon.py]: Do these settings apply for all TOPCON models? If they're tested on a specific device model we'd rather err on the side of caution and define it more narrowly.

File: totalopenstation/models/__init__.py [hunk 3]
```diff
@@ -116,6 +116,7 @@ def run(self):
     'zeiss_elta_r55': ('zeiss_elta_r55', 'ModelConnector', 'Zeiss Elta R55'),
     'nikon_npl_350': ('nikon_npl_350', 'ModelConnector','Nikon NPL 350'),
     'leica_tcr_705': ('leica_tcr_705', 'ModelConnector', 'Leica TCR 705'),
-    'trimble': ('trimble', 'ModelConnector', 'Trimble'),
+    'trimble': ('trimble', 'ModelConnector', 'Trimble'), 
+    'topcon': ('topcon', 'ModelConnector', 'TOPCON'),
```
Comment (steko) [inline totalopenstation/models/__init__.py]: According to their official website the capitalization is "Topcon"

**Review by steko — CHANGES_REQUESTED** [review]
A few minor changes are needed.

File: totalopenstation/models/topcon_gpt_3005.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+# filename: topcon gpt 3005.py
+# Copyright 2008 Enzo Cocca <enzo.ccc@gmail.com>
```
Comment (steko) [inline totalopenstation/models/topcon_gpt_3005.py]: It's 2021 😉

Comment (enzococca) [reply totalopenstation/models/topcon_gpt_3005.py]: done

**Comment by steko** [comment]
All looks good now, I don't understand why one of the travis checks is not working. Any idea @psolyca ?

**Review by steko — APPROVED** [review]

**Review by steko — APPROVED** [review]

**Comment by psolyca** [comment]
> All looks good now, I don't understand why one of the travis checks is not working. Any idea @psolyca ?

@steko No idea, I have seen the message but did not had time to check why.

**Comment by steko** [comment]
I fixed the Travis issue in the meantime, it was a leftover from the old settings. Now there is one check only (since it was basically a duplicate check).

Il 18 settembre 2021 09:13:24 CEST, Damien Gaignon ***@***.***> ha scritto:
>> All looks good now, I don't understand why one of the travis checks is not working. Any idea @psolyca ?
>
***@***.*** No idea, I have seen the message but did not had time to check why.

**Pull request merged**